src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2687,6 +2687,10 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                         Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, distinguisher.First, distinguisher.Second);
                     }
                 }
+                else if (sourceType.SpecialType == SpecialType.System_Void)
+                {
+                    Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, sourceType.ToDisplayString(), targetType.ToDisplayString());
```

Comment:
It is intentional that this new case for reporting ERR_NoImplicitConv uses the same eager formatting and formatting options as the other cases reported in the binder. I do not intend to change that with this change.